pkg/validation/providers.go

```diff
@@ -0,0 +1,67 @@
+package validation
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	"github.com/oauth2-proxy/oauth2-proxy/v7/pkg/apis/options"
+)
+
+// validateMultipleProviders is the initial validation migration for multiple providrers
+// It currently includes only logic which can verify the providers one by one and does not break the valdation pipe
+func validateMultipleProviders(o *options.Options) []string {
+	msgs := []string{}
+
+	if len(o.Providers) == 0 {
+		msgs = append(msgs, "at least one providers has to be defined")
```

Comment:
Nit: This reads better like this I think, not sure if it breaks the logical flow though?
```suggestion
		msgs = append(msgs, "at least one provider has to be defined")
```